processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,375 @@
+		}
+		return &metricspb.Point_Int64Value{Int64Value: intVal}, nil, nil
+	case metricspb.MetricDescriptor_GAUGE_DOUBLE, metricspb.MetricDescriptor_CUMULATIVE_DOUBLE:
+		for _, p := range points {
+			if aggrType == Sum || aggrType == Average {
+				doubleVal += p.GetDoubleValue()
+			} else if aggrType == Max {
+				if p.GetDoubleValue() <= doubleVal {
+					continue
+				}
+				doubleVal = p.GetDoubleValue()
+			}
+
+		}
+		if aggrType == Average {
+			doubleVal /= float64(len(points))
+		}
+		return nil, &metricspb.Point_DoubleValue{DoubleValue: doubleVal}, nil
+	case metricspb.MetricDescriptor_GAUGE_DISTRIBUTION, metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION:
+		if aggrType != Sum {
+			break
+		}
+		boundsToPoints := mtp.groupPointsByBounds(points)
+		for _, v := range boundsToPoints {
+			var distVal *metricspb.DistributionValue
+			for _, p := range v {
+				if distVal == nil {
+					distVal = p.GetDistributionValue()
+					continue
+				}
+				distVal = mtp.computeDistVals(distVal, p.GetDistributionValue())
+			}
+			distVals = append(distVals, &metricspb.Point_DistributionValue{DistributionValue: distVal})
+		}
+		return nil, nil, distVals
+
+	}
+	return nil, nil, nil
+}
+
+// groupPointsByBounds groups distribution value points by the bounds to further determine the aggregatability because distribution value points can only be aggregated if they have match bounds
+// Returns a map with groups of aggregatable distribution value points
+func (mtp *metricsTransformProcessor) groupPointsByBounds(points []*metricspb.Point) map[string][]*metricspb.Point {
+	boundsToPoints := make(map[string][]*metricspb.Point)
+	for _, p := range points {
+		boundsKey := mtp.boundsToString(p.GetDistributionValue().BucketOptions.GetExplicit().Bounds)
+		if groupedPoints, ok := boundsToPoints[boundsKey]; ok {
+			boundsToPoints[boundsKey] = append(groupedPoints, p)
+		} else {
+			boundsToPoints[boundsKey] = []*metricspb.Point{p}
+		}
+	}
+	return boundsToPoints
+}
+
+// boundsToString converts bounds slice into a string
+// Returns a string that represents this bounds
+func (mtp *metricsTransformProcessor) boundsToString(bounds []float64) string {
+	var boundsKey string
+	for _, b := range bounds {
+		boundsKey += fmt.Sprintf("%f-", b)
+	}
+	return boundsKey
+}
+
+// sumOfSquaredDeviation calculates the combined SumOfSquaredDeviation from the ones in val1 and val2
+// Formula derived and extended from https://math.stackexchange.com/questions/2971315/how-do-i-combine-standard-deviations-of-two-groups
+// SSDcomb = SSD1 + n(ave(x) - ave(z))^2 + SSD2 +n(ave(y) - ave(z))^2
+func (mtp *metricsTransformProcessor) computeDistVals(val1 *metricspb.DistributionValue, val2 *metricspb.DistributionValue) *metricspb.DistributionValue {
+	buckets := make([]*metricspb.DistributionValue_Bucket, len(val1.Buckets))
+	for i := range buckets {
+		buckets[i] = &metricspb.DistributionValue_Bucket{
+			Count: val1.Buckets[i].Count + val2.Buckets[i].Count,
```

Comment:
Do you need to preserve exemplars here as well?

Reply:
I am a bit confused about what the exemplar means here conceptually. Is it just a sample point? If so, can I just pick one to preserve arbitrarily?

Reply:
I believe you're correct, and if we are merging buckets, then picking one at random seems like the right thing to do.

Reply:
Make sure you do pick the exemplar randomly though. You don't want to always pick from the same bucket otherwise your "representative example" won't actually be representative of the merged bucket range.